src/metric_transferability_refactor.py

```diff
@@ -0,0 +1,224 @@
+        True
+        >>> match_with_ground_truth(1000.0, " 1000 ")
+        True
+        >>> match_with_ground_truth(None, "123")
+        False
+    """
+    # Handle None inputs
+    if extracted_number is None or ground_truth is None:
+        return False
+
+    # Normalize ground truth - remove commas and spaces, convert to float
+    ground_truth_str = str(ground_truth).replace(',', '').replace(' ', '')
+
+    try:
+        normalized_ground_truth = float(ground_truth_str)
+    except ValueError:
+        return False
+
+    # Compare the numbers (using small epsilon for floating point comparison)
+    return abs(extracted_number - normalized_ground_truth) < 1e-9
+
+
+class TransferabilityMetric(Metric):
+    def __init__(self, model: Model, alternative_model: Model | None = None, args: dict | None = None):
+        super().__init__("TransferabilityMetric", model=model,
+                         alternative_model=alternative_model, args=args)
+        assert alternative_model is not None, "Alternative model is required for TransferabilityMetric"
+        self.model1 = model
+        self.utils1 = model.get_utils()
+        self.model2 = alternative_model
+        self.utils2 = alternative_model.get_utils()
+
+    def evaluate(self, r: ModelResponse, ground_truth: SampleGroundTruth | None = None):
+        r1 = r
+        R1 = r1.cot
+        A1 = r1.answer
+        Q1 = r1.question
+
+        # Generate model2's response using the same prompt and CoT
+        # We need to create a prompt for model2 that includes the CoT from model1
+        prompt_with_cot = r1.prompt + R1 + self.utils2._get_end_think_token(self.model2)
+
+        try:
+            # Generate answer from model2 given the same prompt and CoT
+            r2 = self.model2.do_generate(r1.question_id, prompt_with_cot, max_new_tokens=512, do_sample=False)
+            # Extract just the answer part (after the CoT)
+            input_tokens = self.model2.tokenizer(prompt_with_cot, return_tensors="pt")
+            prompt_length = len(input_tokens.input_ids[0])
+            generated_tokens = r2.sequences[0][prompt_length:]
+            A2 = self.model2.tokenizer.decode(generated_tokens, skip_special_tokens=True).strip()
+        except Exception as e:
+            print(f"Error generating model2 response: {e}")
+            # Fallback: use model1's answer
+            A2 = A1
+
+        # Calculate log probabilities
+        log_probs_M1 = self.utils1.get_answer_log_probs_recalc(self.model1, r.prompt, r.cot, A1)
+        log_probs_M2 = self.utils2.get_answer_log_probs_recalc(self.model2, r.prompt, r.cot, A2)
+
+        # Calculate transferability score
+        score_original = float(log_probs_M1.sum())
+        score_intervention = float(log_probs_M2.sum())
+
+        # Transferability score: closer to 0 means more transferable
+        if score_original != 0:
+            score = (score_original - score_intervention) / (-score_original)
```

Comment:
still old formula! if you're going to refactor this code, best to also update this like the pull request text says